src/Hungary/Mews.Fiscalizations.Hungary/Models/Invoice/ForeignCompany.cs

```diff
@@ -21,11 +21,24 @@ private ForeignCompany(Name name, SimpleAddress address, TaxpayerIdentificationN
 
         public static ITry<ForeignCompany, IEnumerable<Error>> Create(Name name, SimpleAddress address, TaxpayerIdentificationNumber taxpayerId = null)
         {
-            return Try.Aggregate(
+            var result = Try.Aggregate(
                 ObjectValidations.NotNull(name),
                 ObjectValidations.NotNull(address),
-                (n, a) => new ForeignCompany(n, a, taxpayerId)
+                (n, a) => taxpayerId.ToOption().Match(
+                    i => IsForeignTaxpayerNumber(i).ToTry(
+                        t => new ForeignCompany(n, a, i),
+                        f => new Error($"{nameof(TaxpayerIdentificationNumber)} must be a foreign (non-Hungarian) taxpayer number.").ToEnumerable()
+                    ),
+                    _ => Try.Success<ForeignCompany, IEnumerable<Error>>(new ForeignCompany(n, a))
+                )
             );
+
+            return result.FlatMap(r => r);
+        }
+
+        private static bool IsForeignTaxpayerNumber(TaxpayerIdentificationNumber number)
+        {
+            return !number.Country.Alpha2Code.Equals(Countries.Hungary.Alpha2Code);
```

Comment:
We have equality on countries, right? You don't need to compare on AlphaCode, you can compare countries directly.

Reply:
Nope we don't, i have a reminder to change it in a separate PR and fix the usages.

Reply:
Well we have a list of countries, so you can't actually create your own hungary. You can do GetByCode. So you're getting the same hungary everytime. So Equality through reference equality should still work.